habitat_sim/utils/data/dataextractor.py

```diff
@@ -0,0 +1,371 @@
+        self.labels = set(labels)
+        self.cfg = self._config_sim(self.scene_filepath, img_size)
+
+        if sim is None:
+            sim = habitat_sim.Simulator(self.cfg)
+        else:
+            # If a sim is provided we have to make a new cfg
+            self.cfg = self._config_sim(sim.config.sim_cfg.scene.id, img_size)
+            sim.reconfigure(self.cfg)
+
+        self.sim = sim
+        self.pixels_per_meter = 0.1
+        ref_point = self._get_pathfinder_reference_point(self.sim.pathfinder)
+        self.tdv = TopdownView(self.sim, ref_point[1], self.pixels_per_meter)
+        self.topdown_view = self.tdv.topdown_view
+
+        self.pose_extractor = PoseExtractor(
+            self.topdown_view, self.sim.pathfinder, self.pixels_per_meter
+        )
+        self.poses = self.pose_extractor.extract_poses(
+            labels=self.labels
+        )  # list of poses
+        self.label_map = {0.0: "unnavigable", 1.0: "navigable"}
+
+        # Configure the output each data sample
+        self.out_name_to_sensor_name = {
+            "rgba": "color_sensor",
+            "depth": "depth_sensor",
+            "semantic": "semantic_sensor",
+        }
+        self.output = output
+
+    def __len__(self):
+        return len(self.poses)
+
+    def __getitem__(self, idx):
+        if isinstance(idx, slice):
+            start, stop, step = idx.start, idx.stop, idx.step
+            if start is None:
+                start = 0
+            if stop is None:
+                stop = len(self.poses)
+            if step is None:
+                step = 1
+
+            return [
+                self.__getitem__(i)
+                for i in range(start, stop, step)
+                if i < len(self.poses)
+            ]
+
+        pos, rot, label = self.poses[idx]
+        new_state = AgentState()
+        new_state.position = pos
+        new_state.rotation = rot
+        self.sim.agents[0].set_state(new_state)
+        obs = self.sim.get_sensor_observations()
+        sample = {
+            out_name: obs[self.out_name_to_sensor_name[out_name]]
+            for out_name in self.output
+        }
+        sample["label"] = self.label_map[label]
+
+        return sample
+
+    def close(self):
+        r"""Deletes the instance of the simulator. Necessary for instatiating a different ImageExtractor.
+        """
+        self.sim.close()
+        del self.sim
+
+    def _config_sim(self, scene_filepath, img_size):
+        sim_settings = {
+            "width": img_size[1],  # Spatial resolution of the observations
+            "height": img_size[0],
+            "scene": scene_filepath,  # Scene path
+            "default_agent": 0,
+            "sensor_height": 1.5,  # Height of sensors in meters
+            "color_sensor": True,  # RGBA sensor
+            "semantic_sensor": True,  # Semantic sensor
+            "depth_sensor": True,  # Depth sensor
+            "silent": True,
+        }
+
+        return make_cfg(sim_settings)
```

Comment:
Using the dictionary and helper function doesn't really seem necessary here.  Just making the config in this method would be cleaner IMO.

Reply:
For the next PR im working on that depends on this one, I need to call this method from multiple classes, but for this branch it does seem cleaner to do what you suggested.